content/posts/introducing-slight.md

```diff
@@ -0,0 +1,155 @@
+
+## How does it work?
+
+You might noticed that slight takes two arguments. The first argument is a configuration file called `slightfile.toml`. The second argument is a WebAssembly module. They represent two phases of software development: operation and development, resp.
+
+### Operations
+Let's take a look at the `slightfile.toml` file:
+
+```toml
+specversion = "0.2"
+
+[[capability]]
+resource = "kv.filesystem"
+name = "my-container"
+```
+
+The slight configuration file lists the capabilities that the application needs. In this case, the application needs a key-value store called `my-container`. The key-value store is a filesystem-based key-value store.
+
+Without changing the code, you can change the key-value store to a different infrastructure by changing the configuration file. For example, you can change the key-value store to a Redis-based key-value store by changing the configuration file to the following:
+
+```toml
+specversion = "0.2"
+
+[[capability]]
+resource = "kv.redis"
+name = "my-container"
+    [capability.configs]
+    REDIS_ADDRESS = "redis://127.0.0.1:6379"
+```
+
+Or you can change the key-value store to a Azure BlobStorage-based key-value store by changing the configuration file to the following:
+
+```toml
+specversion = "0.2"
+
+[[capability]]
+resource = "kv.azblob"
+name = "my-container"
+    [capability.configs]
+    AZURE_STORAGE_ACCOUNT = "${azapp.AZURE_STORAGE_ACCOUNT}"
+    AZURE_STORAGE_KEY = "${azapp.AZURE_STORAGE_KEY}"
+```
+
+### Development
+
+If you open the `src/main.rs` file, you will see the following code:
+
+```rust
+use kv::*;
+wit_bindgen_rust::import!("wit/kv_v0.1.0/kv.wit");
+```
+
+The application code imports the key-value store interface from the `wit/kv_v0.1.0/kv.wit` file. It **does not** know, at dev time, which key-value store is used. All it knows is that it can use a few operations to interact with the key-value store.
+
+You may wonder what is the `wit/kv_v0.1.0/kv.wit` file. It is a WIT file that defines the interface of the key-value store. It looks like the following: 
+
+```go
+// A key-value store interface.
+
+resource kv {
+	// open a key-value store
+	static open: func(name: string) -> expected<kv, error>
+
+	// get the payload for a given key.
+	get: func(key: string) -> expected<payload, error> 
+
+	// set the payload for a given key.
+	set: func(key: string, value: payload) -> expected<unit, error>
+
+	// delete the payload for a given key.
+	delete: func(key:string) -> expected<unit, error>
+}
+```
+
+Only at runtime, the WebAssembly runtime will provide the implementation of the key-value store interface. At runtime, slight knows which key-value store is used by looking at the configuration file.
+
+## Standardization
+
+Without standardization and community support, SpiderLightning will not go far. We are working with the WebAssembly community to standardize these distributed application capability interfaces.
```

Comment:
Try not to focus on SpiderLightning's success. The success that we want to see more than anything is the success of the community around the interfaces. Success via standardization will provide a common set of capabilities accessible by ANY runtime that exposes those interfaces.